ARMmbed/mbed-os
# Add a new top level config.h for mbedtls

## Description
This PR enables the possibility to use a new configuration file `config-no-entropy.h` recently added to mbed TLS (https://github.com/ARMmbed/mbedtls/pull/615) that allows the subset of the library to be used when the target MCU does not implement an entropy source. To do so, a `ENTROPY_SOURCE` value is added to the targets that have entropy source in the `target.json` file. Also, the mbed TLS importer script is modified to copy a `config-no-entropy.h` into the mbedtls include folder.

I have assumed that all targets that define `MBEDTLS_ENTROPY_HARDWARE_ALT` implement an entropy source, so I have added `ENTROPY_SOURCE` to all of these.

## Status
IN DEVELOPMENT

## Migrations
NO

## Related PRs
This is related to the PR https://github.com/ARMmbed/mbedtls/pull/615 in mbed TLS.

## Steps to test or reproduce
* Update the feature mbedtls by navigating to the feature/mbedtls/importer and running the following sequence of commands:
```
make clean
make update
make
``` 
* Run the tests as usual or use this version of mbed OS to build any application. Applications that use mbed TLS features that require an entropy source, but the target MCU does not offer one will fail to compile. Otherwise, the applications should work as normal.
